File: src/execgo_playground/adapters/catalog.py
from __future__ import annotations

from .autogen_adapter import AutoGenAdapter
from .crewai_adapter import CrewAIAdapter
from .langgraph_adapter import LangGraphAdapter


def list_adapters() -> dict[str, type]:
    return {
        "langgraph": LangGraphAdapter,
        "crewai": CrewAIAdapter,
        "autogen": AutoGenAdapter,
    }


def get_adapter(name: str):
    catalog = list_adapters()
    try:
        return catalog[name.lower()]()
    except KeyError as exc:
        raise ValueError(f"unknown adapter {name}") from exc

File: src/execgo_playground/adapters/catalog.ts
import { AutoGenAdapter } from "./autogen_adapter.js";
import { BaseAdapter } from "./base.js";
import { CrewAIAdapter } from "./crewai_adapter.js";
import { LangGraphAdapter } from "./langgraph_adapter.js";

export function listAdapters(): Record<string, new () => BaseAdapter> {
  return {
    langgraph: LangGraphAdapter,
    crewai: CrewAIAdapter,
    autogen: AutoGenAdapter,
  };
}

export function getAdapter(name: string): BaseAdapter {
  const Adapter = listAdapters()[name.toLowerCase()];
  if (!Adapter) {
    throw new Error(`unknown adapter ${name}`);
  }
  return new Adapter();
}
